cpp/include/cudf/strings/detail/split_utils.cuh

```diff
@@ -22,6 +22,31 @@ namespace cudf {
 namespace strings {
 namespace detail {
 
+constexpr bool is_whitespace(char_utf8 ch) { return ch <= ' '; }
+
+struct whitespace_token_counter_fn {
+  __device__ size_type count_tokens(string_view d_str) const
```

Comment:
I see a little bit awkward when using this struct:
```
  whitespace_token_counter_fn counter{max_tokens};
  return counter.count_tokens(d_str);
```

Why not just make this a device function like `count_whitespace_token(...)`? Then we can just use this:
```
  return count_whitespace_token(d_str, max_tokens);
```

Reply:
It needs to maintain state across calls.